lib/msf/core/exploit/pgadmin.rb

```diff
@@ -10,17 +10,29 @@ module Msf
   module Exploit::PgAdmin
     include Msf::Exploit::Remote::HttpClient
 
-    def get_version
-      res = send_request_cgi('uri' => normalize_uri(target_uri.path, 'login'), 'keep_cookies' => true)
-      return unless res&.code == 200
+    def auth_required?
+      res = send_request_cgi('uri' => normalize_uri(target_uri.path), 'keep_cookies' => true)
+      if res&.code == 302 && res.headers['Location']['login']
+        true
+      elsif res&.code == 302 && res.headers['Location']['browser']
+        false
+      end
```

Comment:
```suggestion
      if res&.code == 302
        return true if res.headers['Location']['login']
        return false if res.headers['Location']['browser']
      end
```

Also, it would be nice to have an explicit return value should all the conditions fail. It